srtcore/buffer.cpp

```diff
@@ -1590,12 +1588,15 @@ int CRcvBuffer::readMsg(char* data, int len, ref_t<SRT_MSGCTRL> r_msgctl)
     msgctl.pktseq = pkt1.getSeqNo();
     msgctl.msgno = pkt1.getMsgSeq();
 
+    SRT_ASSERT(len > 0);
     int rs = len;
     while (p != (q + 1) % m_iSize)
     {
-        int unitsize = (int) m_pUnit[p]->m_Packet.getLength();
-        if ((rs >= 0) && (unitsize > rs))
-            unitsize = rs;
+        const int pktlen = (int)m_pUnit[p]->m_Packet.getLength();
+        if (pktlen > 0)
+            countBytes(-1, -pktlen, true);
+
+        const int unitsize = ((rs >= 0) && (unitsize > rs)) ? rs : pktlen;
 
         if (unitsize > 0)
         {
```

Comment:
What will then happen in the line below, if the memory assigned to `data` are of less size than `pktlen` and the value passed as `len` to `readData` is -1? And if we don't predict it be -1, why is then `rs >= 0` checked?

I know that it wasn't subject of this PR, but it looks weird. The condition should prevent `memcpy` to be called with a size that exceeds the allocated memory.